osquery/distributed/distributed.cpp

```diff
@@ -80,73 +100,173 @@ Status Distributed::pullUpdates() {
   return Status(0, "OK");
 }
 
+/**
+ * getPendingQueryCount
+ * Return number of queries waiting to be executed.
+ */
 size_t Distributed::getPendingQueryCount() {
-  std::vector<std::string> queries;
-  scanDatabaseKeys(kQueries, queries, kDistributedQueryPrefix);
-  return queries.size();
+  int num = 0;
+  for (auto item : results_) {
+    if (item.isPending()) {
+      num += 1;
+    }
+  }
+  return num;
 }
 
-size_t Distributed::getCompletedCount() {
-  return results_.size();
+// Returns the number of time distributed_read endpoint was accessed
+size_t Distributed::numDistReads() {
+  return numDistReads_;
 }
 
-Status Distributed::serializeResults(std::string& json) {
+// Returns the number of time distributed_write endpoint was accessed
+size_t Distributed::numDistWrites() {
+  return numDistWrites_;
+}
+
+Status Distributed::serializeResults(std::string& json,
+                                     DistributedQueryResult* item) {
```

Comment:
Instead of a pointer could a const reference be passed?

Reply:
item can be null, pointer comparison used in function to serialize a single or all results.